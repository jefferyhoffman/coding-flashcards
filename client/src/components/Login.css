.Login {
  height: 100vh;
  background-image: url(https://picsum.photos/2000/1000?blur);
  position: relative;
}

.Login #firebaseui_container {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.Login .firebaseui-page-provider-sign-in {
  background-color: #fff;
  background-color: rgba(255, 255, 255, 0.5);
  border-radius: 5px;
}
