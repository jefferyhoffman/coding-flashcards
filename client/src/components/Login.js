import React, { Component } from 'react';
import { FirebaseAuth } from 'react-firebaseui';
import PropTypes from 'prop-types';
import './Login.css';

class Login extends Component {
  static propTypes = {
    firebase: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);

    this.uiConfig = {
      // Popup signin flow rather than redirect flow.
      signInFlow: 'popup',
      // Redirect to /signedIn after sign in is successful.
      signInSuccessUrl: '/dashboard',
      // We will display Google and GitHub as auth providers.
      signInOptions: [
        props.firebase.auth.GithubAuthProvider.PROVIDER_ID,
        props.firebase.auth.GoogleAuthProvider.PROVIDER_ID,
        props.firebase.auth.EmailAuthProvider.PROVIDER_ID
      ]
    };
  }

  render() {
    return (
      <div className="Login">
        <FirebaseAuth
          uiConfig={this.uiConfig}
          firebaseAuth={this.props.firebase.auth()}
        />
      </div>
    );
  }
}

export default Login;
