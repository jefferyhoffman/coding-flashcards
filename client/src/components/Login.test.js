import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
    it('renders correctly', () => {
        const login = shallow(<Login />);
        
        expect(login).toMatchSnapshot();
    });
});
