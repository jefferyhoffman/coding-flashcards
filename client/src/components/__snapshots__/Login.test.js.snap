// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> renders correctly 1`] = `
<div
  className="Login"
>
  <FirebaseAuth
    firebaseAuth={
      Object {
        "apiKey": "AIzaSyDKxUZI84h3nLirimQuw11Vy2mmEC56GV0",
        "appName": "[DEFAULT]",
        "authDomain": "coding-flashcards.firebaseapp.com",
        "currentUser": null,
      }
    }
    uiConfig={
      Object {
        "signInFlow": "popup",
        "signInOptions": Array [
          "github.com",
          "google.com",
          "password",
        ],
        "signInSuccessUrl": "/dashboard",
      }
    }
  />
</div>
`;
